capstone/src/main/webapp/js/autocomplete.js

```diff
@@ -0,0 +1,140 @@
+const inp = document.getElementById('search');
+const currentFocus = -1;
+/**
+ * @param {Object} autocompleteListElmt
+ * @return {bool}
+ * Adds the 'active' tag to an autocomplete elmt
+ */
+function addActiveTag(autocompleteListElmt) {
+  if (!autocompleteListElmt) {
+    return false;
+  }
+  removeActiveTag(autocompleteListElmt);
+  if (currentFocus >= autocompleteListElmt.length) {
+    currentFocus = 0;
+  } else if (currentFocus < 0) {
+    currentFocus = (autocompleteListElmt.length - 1);
+  }
+  autocompleteListElmt[currentFocus].classList.add('autocomplete-active');
+}
+
+/**
+ * @param {Object} autocompleteListElmt
+ * @return {void}
+ * Removes the 'active' tag from an autocomplete elmt
+ */
+function removeActiveTag(autocompleteListElmt) {
+  for (listElmt in autocompleteListElmt) {
+    if (true) {
+      listElmt.classList.remove('autocomplete-active');
+    }
+  }
+}
+
+/**
+ * @param {Object} elmnt
+ * @param {Object} inp
+ * @return {void}
+ * Closes dropdown autocomplete list(s)
+ */
+function closeAllLists(elmnt, inp) {
+  const autoItems = document.getElementsByClassName('autocomplete-items');
+  for (item in autoItems) {
+    if (elmnt != item && elmnt != inp) {
+      item.parentNode.removeChild(item);
+    }
+  }
+}
+
+/**
+ * @return{Object} arr
+ * Current placeholder until servlet is created
+ */
+function getListOfSuggestions() {
+  const arr = ['Hello', 'Hi', 'Howdy'];
+  return arr;
+}
+
+/**
+ * @param {Object} arr
+ * @param {string} val
+ * @return {Object} trimArr
+ * Identifies and returns relevant suggestions in the arr
+ */
+function getRelevantDataSuggestions(arr, val) {
+  const trimArr = [];
+  for (const arrElt of arr) {
+    if (arrElt.substr(0, val.length).toUpperCase() == val.toUpperCase()) {
+      trimArr.push(arrElt);
+    }
+  }
+  return trimArr;
+}
+
+/**
+ * @param {Object} trimArr
+ * @param {Object} autocompleteList
+ * @return {void}
+ * Takes relevant suggestions and displays them in DOM
+ */
+function displaySuggestions(trimArr, autocompleteList) {
+  for (arrElt in trimArr) {
+    if (true) {
+      const listElmt = document.createElement('DIV');
+      listElmt.innerHTML = '<strong>' +
+                           arrElt.substr(0, val.length) +
+                           '</strong>';
+      listElmt.innerHTML += arrElt.substr(val.length);
+      listElmt.innerHTML += '<input type='+'hidden'+'value='+arrElt+'>';
+      listElmt.addEventListener('click', function(e) {
+        inp.value = this.getElementsByTagName('input')[0].value;
+        closeAllLists();
+      });
+      autocompleteList.appendChild(listElement);
+    }
+  }
+}
+
+/* Event occurrance when input is provided to autocomplete field */
+document.addEventListener('input', function(e) {
+  closeAllLists();
+  const val = this.value;
+  if (!val) {
+    return false;
+  }
+  const arr = getListOfSuggestions();
+  currentFocus = -1;
+  const autocompleteList = document.createElement('div');
```

Comment:
can this element be defined in the HTML?

Reply:
It could-however, since the id and the class are both added in the program, I'm not sure how to uniquely identify the autocompleteList in order to call it in the javascript function?

Reply:
sounds good! doing that now :)

Reply:
+1 to element being defined in html. Fundamentally we have to lists in this html, a list of suggestions, and a list of selected colleges. having them both in html helps the reader a lot.